src/server.js

```diff
@@ -365,11 +367,11 @@ function initializeClient(socket, client, generateToken, token) {
 				}
 			});
 
-			sendInitEvent(token);
-		});
-	} else {
-		sendInitEvent(null);
+			return newToken;
+		}).catch((error) => log.error(error));
 	}
+
+	Promise.resolve(tokenPromise).then((t) => sendInitEvent(t));
```

Comment:
2 cases:

- `tokenPromise` is `null` so `Promise.resolve(null).then(...)` resolves right away
- `tokenPromise` is a promise of `token` so `Promise.resolve(tokenPromise).then(...)` resolves when `tokenPromise` resolves (because spec allows for `Promise.resolve([a promise])`.